### DISCUSSION TITLE
How to read peak force

### DISCUSSION THREAD

**Discussion by FGarelli** [OpenRadioss/OpenRadioss#1761] [Q&A] [answered]
Hello everyone,
I'm using Paraview and Octave for the post processing but the way of how the time history output file is written is not very clear.
I'm requesting through a LS-Dyna input file and with these keywords: *DATABASE_SECFORC, *DATABASE_SPCFORC  and *DATABASE_RBDOUT the force during the crash test. I'm trying to measure the force in the boundary, in the cross section that i created and of the reaction on the rigid body that is loading the part.
How can I get the plot of the forces and how I can understand the variables written on the .th file?
I attached the input file and the .th of the results.
[crash_test_A00.zip](https://github.com/OpenRadioss/OpenRadioss/files/12251190/crash_test_A00.zip)

**Comment by FGarelli** [comment]
These are the plot that i can get from paraview but i cannot understand which comoponent are and the spc force should be more then 0, it should be as the rigid force more or less
![spc_force](https://github.com/OpenRadioss/OpenRadioss/assets/132582318/1cd996e9-9eac-4822-98e2-cd3e098f19a8)
![rigid_force](https://github.com/OpenRadioss/OpenRadioss/assets/132582318/f95eb0dd-be4d-4c26-a999-60ffc1bc1ecc)
![section_force](https://github.com/OpenRadioss/OpenRadioss/assets/132582318/a180f468-8c7b-4da7-83a3-2151f13f3bf9)

**Comment by MarianBulla** [comment, marked as answer]
Hello FGarelli,
the output in the T01 file for forces is done as momentum (impulse). So you have to calculate the derivative of these vales to get to force (here, done manually in Excel):
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/716704d8-34cd-4b4c-bfab-506e016934be)

With your units, one obtain the force in X-axis in [kN] vs. time [ms] plot.

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by FGarelli** [reply]
Thank you Marian! I didn't know. For the energy is the same? Or the actual value that i found is the absolute one?

**Reply by MarianBulla** [reply]
Hello FGarelli,
the energies are as they are. No need to adjust or recalculate the values:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/f2b16e91-cb5b-491c-9ba7-4976ebcae4d5)

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by MarianBulla** [comment]
There are some values which are tracked by OpenRadioss as they are, and some which have to be re-computed:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/e34c4f71-1f6f-42de-88e2-5bd87255bfe9)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/15a86b37-2501-4ffb-bfe6-45452c46de19)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/b73cbe4f-b727-429e-9025-3032054e901a)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/a376abce-00eb-47db-b66e-8dbd8d0995b8)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/d02b687a-36d5-41b2-abfa-3c2d552cb30b)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/d8578424-3081-401c-8bc7-a55cbd25b730)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/895f3bbc-bf63-4db3-85ad-940c04f33963)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/ed5be3db-5ca0-463e-b7fb-bf064ee57620)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/90aa51dc-7e36-4370-b076-2eefd45487aa)

So, for some results, you have to double check or review in the code, how it is stored in the T01 file and compute the neccessary values.

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by FGarelli** [reply]
Thank you Marian, all clear now!!

**Reply by li2qn** [reply]
Could you please direct me to the like where to find the list above? Thanks

**Reply by li2qn** [reply]
I was trying to calculate the reaction forces at node, and request RAECX in TH. I always have the same issue with the output values.

**Reply by MarianBulla** [reply]
Hi Lee,
Did you convert the T01 file into cvs format, where it is stored in a human-readable format and reviewed the content ?

**Reply by MarianBulla** [reply]
Hi Lee,
Did you convert the T01 file into cvs format, where it is stored in a human-readable format and reviewed the content ?

**Reply by li2qn** [reply]
Yes. I have converted the T01 files to a csv

**Reply by MarianBulla** [reply]
Do you see the values for the RAECX forces ?

**Reply by li2qn** [reply]
How is actually the node reaction forces being calculated in T01 file? I saw you upload some of the table/list above? where to find them?

**Reply by MarianBulla** [reply]
This list is an old list, which we don't maintain any more.
So this list will not help you.
Can you add your T01 file, so I can have a direct look into it, please ?
(or a smaller model, using this feature)

**Reply by li2qn** [reply]
Yes. I do see the values for RAECS forces. should be something like below right? and the column O should be the RAECX correct?
![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/2c25e25e-3d95-487e-8f95-814cca441abd)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/e6f7acf2-2ce2-46d2-864a-dace54ebc17d)

**Reply by MarianBulla** [reply]
Yes, exactly. Maybe you have to create the 1st derivative by time with this signal to get the forces. 
Do you have access to YouTube ?
In our YT channel, I'm showing in a video how to postprocess the forces from the csv file, using excel.

**Reply by li2qn** [reply]
OK, Yes. Thanks so much. Here is the files. Thanks for your time and help
[propeller_s4r_s4r_1storder_f_radioss_0000.zip](https://github.com/OpenRadioss/OpenRadioss/files/12895404/propeller_s4r_s4r_1storder_f_radioss_0000.zip)

**Reply by li2qn** [reply]
Yes. I have access to YouTube, could you please direct me to the videos?

**Reply by MarianBulla** [reply]
https://www.youtube.com/watch?v=EPZenaaDDak&t=42s

Starting from the 5th minute and further...

**Reply by li2qn** [reply]
Many thanks. could you please show me how you acutely calculate the reaction forces in the above example? That would be super

**Reply by MarianBulla** [reply]
ok, you can also request the forces acting on the RigidBody primary node. I think they will be more or less the same, by adding this to your timehistory request: (the lower part is new)
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
#- 14. TIME HISTORIES:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/TH/NODE/1
node 90001
#     var1      var2      var3      var4      var5      var6      var7      var8      var9     var10
       VRZ     REACX     REACY     REACZ    REACXX    REACYY    REACZZ  
#    NODid     Iskew                                           NODname
     90001 
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/TH/RBODY/2
TH RBODY
#     var1      var2      var3      var4      var5      var6      var7      var8      var9     var10
DEF       
#     Obj1      Obj2      Obj3      Obj4      Obj5      Obj6      Obj7      Obj8      Obj9     Obj10
         1
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|

![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/b3b68fc0-ad5a-4a5f-8ace-dece2c97a539)

**Reply by MarianBulla** [reply]
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/016b6946-bc42-4050-b8e2-2b760bf9862c)

**Reply by MarianBulla** [reply]
This is what you will see as resultant force:

![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/1fafea11-98ff-4550-8a76-fd77ee1f7354)

**Reply by li2qn** [reply]
So that means all the forces calculated in TH is the derivative vs time?

**Reply by MarianBulla** [reply]
Yes, exactly!
We store the impulse in the T01 file. 
In the YouTube, I'm showing how to get the force by simple derivating the impulse signal by time.

**Reply by li2qn** [reply]
Thanks so much. That means If i get the force by simple derivating the impulse signal by time, the value should be more or less similar to what I get from the output of RBODY?

**Reply by li2qn** [reply]
One more question regarding the plot you shared for resultant force. how did you get the plot? is that a variable able to directly request from TH/RBODY or need to be post processed as well? thanks

**Reply by li2qn** [reply]
and also I was trying to compare the results which I get from abaqus, looks like the forces are 10 times lower than what I have in abaqus. In terms of energy it look correct. Could you please help me check what is going on here. I update the abaqus odb and input file as well as the .csv file for this model we discussed here 
[propeller_abaqus_radioss.zip](https://github.com/OpenRadioss/OpenRadioss/files/12897705/propeller_abaqus_radioss.zip)

**Reply by MarianBulla** [reply]
Yes, regarding the forces from the RigidBody and the main nodal forces, they should be quite more or less the same.
I used HyperView to get the resultant plot. The values are computed automatically be the postprocessor.
I don't know, why the forces are so different between Abaqus and OpenRadioss. Maybe there is a difference in the loadcase ?
Should be reviewed closely. 
I will have s closer look, at it and will try to figure out, where it can come from.

**Reply by MarianBulla** [reply]
Regarding your Abaqus results: How do you postprocess them ?
Becasue, if I compare the results from Abaqus and OpenRadioss, I see the similar stress and alomse exactly the same kinetic energy:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/bef1f32d-00c7-4eab-93a9-84c523dc8e2b)

**Reply by li2qn** [reply]
Yes. I also see the similar results in terms of stress and KE. But if you see the history outputs, there are reaction forces stored, the reaction forces are in the order of E+05, while in Radioss there are in the order of E+04

**Reply by MarianBulla** [reply]
Just for my understanding: You are using initial rotational velocity and your structure is quite symmetic. Where should these forces come from and how big should these forces be ?
Because the kinetic energy, rotational velocity, mass and geometry are more or less exactly between these two solvers. Even the stresses are in the same range.

**Reply by li2qn** [reply]
Some info for your understanding, "The propeller is assumed to be spinning in a vacuum with zero initial stresses. Once the stress state due to the centrifugal loading is established, the solution (displacements, stresses, and strains) is expected to oscillate about this state with time. The mass moment of inertia of the propeller about its central axis increases slightly due to the stretching of the blades in the mean stress state under the centrifugal loading." Anyway, let me check the forces on another model I was able to complete previously, will update here soon. Thanks for your kindly help, that helps a lot.

**Reply by li2qn** [reply]
![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/d46b0406-9f15-4271-bd10-5aedfbf80c84)
here is the force plot from ABAQUS

**Reply by MarianBulla** [reply]
Yes indeed, it is a very interesting model.
Regarding the stress waves, please see this example, I add to the materials section:
https://github.com/OpenRadioss/ModelExchange/tree/main/Components/Materials/ElastoPlasticLaw/Law001/Shells/Stress_Wave

I modified it slightly to measure the forces, once the reach to the end:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/96004267-e5e4-4fa0-901b-1994d71728ab)

Here is this slightly modified model:
[beam_002_0000.zip](https://github.com/OpenRadioss/OpenRadioss/files/12925338/beam_002_0000.zip)

Hope it helps you ?

Thanks and best regards,
Marian
